### ISSUE TITLE
auto closing of tags appears to not work on gjs/gts files

### ISSUE THREAD

**Issue by NullVoxPopuli** [lifeart/vsc-ember-syntax#44]
[Screencast from 2023-10-15 09-11-24.webm](https://github.com/lifeart/vsc-ember-syntax/assets/199018/251853fc-bace-4f01-9e2b-8f25af3789a6)

**Comment by evoactivity** [comment]
When I did my deep dive into this to support gts/gjs I was not able to figure this part out. My assumption was injecting the html/handlebars context in a template tag should enable auto closing tags but that doesn't work.

I'll have another look some point next week to.

**Comment by lifeart** [comment]
It may be a case where such cases handled by https://github.com/lifeart/typescript-hbs-plugin

**Comment by evoactivity** [comment]
These docs should be helpful when I get time to look at this
https://code.visualstudio.com/api/language-extensions/embedded-languages

**Comment by IgnaceMaes** [comment]
Not sure where exactly this should be configured in this extension.

Made a PR to `vscode-auto-close-tag` for people relying that: https://github.com/formulahendry/vscode-auto-close-tag/pull/224

To get it working already today, you could install the extension and add this config:

```json5
{
  // other settings ...
  "auto-close-tag.activationOnLanguage": [
    "*",
    "glimmer-js",
    "glimmer-ts",
  ],
}
```

**Comment by evoactivity** [comment]
Interesting approach, will be a lot simplar than adding this to the language server. It might be worth adding that addon to the extension pack rather than here though, especially since your PR was accepted installing it should just work.

**Comment by IgnaceMaes** [comment]
That seems reasonable to me, I can look to PR it. We'll need some info in the README though as to why it's needed, as VS Code ships this natively since https://code.visualstudio.com/updates/v1_16#_html-close-tags.

Seems like a good quick blog post to make as well, if I can find some time 👀

**Comment by evoactivity** [comment]
hmm, this is kind of annoying, it auto closes when I'm writing typescript

<img width="99" alt="image" src="https://github.com/lifeart/vsc-ember-syntax/assets/83799/2dadbc8b-4124-421e-9551-29f1e24eb0a7">

**Comment by IgnaceMaes** [comment]
Good catch, apparently it's a known issue reported 8 years ago 🙃  (https://github.com/formulahendry/vscode-auto-close-tag/issues/17)

I can take a look to fix it in the extension, but at that point it might be worth rather to take over the extension's functionality in this one?

**Comment by lifeart** [comment]
@IgnaceMaes we could do it in our `hbs-typescript-plugin` (it's dependency of this extension)
https://github.com/lifeart/typescript-hbs-plugin/blob/92e1caf5c62ba7a8e0d1124bed7ac566097ad213/src/html-template-language-service.ts#L237

**Comment by evoactivity** [comment]
Using `vscode-html-languageserver` for this is the proper way to do it. I began looking into that with your `typescript-hbs-plugin` but I'm just not sure where to begin with it to be honest. I couldn't even figure out what calls `getJsxClosingTagAtPosition`, I assume something in vscode, but then how do we get it to be called for `<template></template>` regions?

It'd be good to get the full html intellisense working, not just auto closing tags too.
